File: src/main/java/com/citasmedicas/appcitasmedicas/Service/DoctorService.java
package com.citasmedicas.appcitasmedicas.Service;

import com.citasmedicas.appcitasmedicas.dto.Request.CreateDoctorRequest;
import com.citasmedicas.appcitasmedicas.dto.Request.UpdateDoctorRequest;
import com.citasmedicas.appcitasmedicas.dto.Response.DoctorResponse;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface DoctorService {
    
    DoctorResponse create(CreateDoctorRequest req);
    
    DoctorResponse findById(Long id);
    
    List<DoctorResponse> findAll(Pageable pageable);
    
    List<DoctorResponse> findBySpecialty(Long specialtyId, Pageable pageable);
    
    DoctorResponse update(Long id, UpdateDoctorRequest req);
    
    void delete(Long id);
}

File: frontend/src/services/doctorService.js
import api from './api';

export const doctorService = {
  getAll: (page = 0, size = 100, specialtyId = null) => {
    const url = specialtyId
      ? `/doctors?specialtyId=${specialtyId}&page=${page}&size=${size}`
      : `/doctors?page=${page}&size=${size}`;
    return api.get(url);
  },

  getById: (id) =>
    api.get(`/doctors/${id}`),

  create: (data) =>
    api.post('/doctors', data),

  update: (id, data) =>
    api.patch(`/doctors/${id}`, data),

  delete: (id) =>
    api.delete(`/doctors/${id}`),
};